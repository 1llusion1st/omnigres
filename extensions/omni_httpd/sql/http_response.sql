-- NULL status
select omni_httpd.http_response(status => null, body => 'test');

-- NULL headers
select omni_httpd.http_response(headers => null, body => 'test');

-- NULL body
select omni_httpd.http_response(body => null);
select omni_httpd.http_response();

-- Text body
select omni_httpd.http_response(body => 'text');

-- JSON body
select omni_httpd.http_response(body => '{}'::json);

-- JSONB body
select omni_httpd.http_response(body => '{}'::jsonb);

-- Binary body
select omni_httpd.http_response(body => convert_to('binary', 'UTF8'));

-- Specifying status
select omni_httpd.http_response(status => 404);

-- Specifying headers
select
    omni_httpd.http_response(headers => array [omni_httpd.http_header('test', 'value')]::omni_httpd.http_header[],
                             body => null);

-- Merging headers with inferred ones
select
    omni_httpd.http_response(headers => array [omni_httpd.http_header('test', 'value')]::omni_httpd.http_header[],
                             body => 'test');

-- Overriding content type
select
    omni_httpd.http_response(headers => array [omni_httpd.http_header('content-type', 'text/html')], body => 'test');
-- Overriding content type, with a different case
select
    omni_httpd.http_response(headers => array [omni_httpd.http_header('Content-Type', 'text/html')], body => 'test');

--- Shortcut syntax with body first
select omni_httpd.http_response('test');
select omni_httpd.http_response('"test"'::json);
select omni_httpd.http_response('"test"'::jsonb);
