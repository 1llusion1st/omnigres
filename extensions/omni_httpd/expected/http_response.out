-- NULL status
select omni_httpd.http_response(status => null, body => 'test');
                                             http_response                                              
--------------------------------------------------------------------------------------------------------
 omni_httpd.http_response(("\\x74657374",200,"{""(content-type,\\""text/plain; charset=utf-8\\"")""}"))
(1 row)

-- NULL headers
select omni_httpd.http_response(headers => null, body => 'test');
                                             http_response                                              
--------------------------------------------------------------------------------------------------------
 omni_httpd.http_response(("\\x74657374",200,"{""(content-type,\\""text/plain; charset=utf-8\\"")""}"))
(1 row)

-- NULL body
select omni_httpd.http_response(body => null);
           http_response           
-----------------------------------
 omni_httpd.http_response((,200,))
(1 row)

select omni_httpd.http_response();
           http_response           
-----------------------------------
 omni_httpd.http_response((,200,))
(1 row)

-- Text body
select omni_httpd.http_response(body => 'text');
                                             http_response                                              
--------------------------------------------------------------------------------------------------------
 omni_httpd.http_response(("\\x74657874",200,"{""(content-type,\\""text/plain; charset=utf-8\\"")""}"))
(1 row)

-- JSON body
select omni_httpd.http_response(body => '{}'::json);
                               http_response                                
----------------------------------------------------------------------------
 omni_httpd.http_response(("\\x7b7d",200,"{""(content-type,text/json)""}"))
(1 row)

-- JSONB body
select omni_httpd.http_response(body => '{}'::jsonb);
                               http_response                                
----------------------------------------------------------------------------
 omni_httpd.http_response(("\\x7b7d",200,"{""(content-type,text/json)""}"))
(1 row)

-- Binary body
select omni_httpd.http_response(body => convert_to('binary', 'UTF8'));
                                           http_response                                           
---------------------------------------------------------------------------------------------------
 omni_httpd.http_response(("\\x62696e617279",200,"{""(content-type,application/octet-stream)""}"))
(1 row)

-- Specifying status
select omni_httpd.http_response(status => 404);
           http_response           
-----------------------------------
 omni_httpd.http_response((,404,))
(1 row)

-- Specifying headers
select
    omni_httpd.http_response(headers => array [omni_http.http_header('test', 'value')]::omni_http.http_header[],
                             body => null);
                     http_response                     
-------------------------------------------------------
 omni_httpd.http_response((,200,"{""(test,value)""}"))
(1 row)

-- Merging headers with inferred ones
select
    omni_httpd.http_response(headers => array [omni_http.http_header('test', 'value')]::omni_http.http_header[],
                             body => 'test');
                                                      http_response                                                      
-------------------------------------------------------------------------------------------------------------------------
 omni_httpd.http_response(("\\x74657374",200,"{""(test,value)"",""(content-type,\\""text/plain; charset=utf-8\\"")""}"))
(1 row)

-- Overriding content type
select
    omni_httpd.http_response(headers => array [omni_http.http_header('content-type', 'text/html')], body => 'test');
                                 http_response                                  
--------------------------------------------------------------------------------
 omni_httpd.http_response(("\\x74657374",200,"{""(content-type,text/html)""}"))
(1 row)

-- Overriding content type, with a different case
select
    omni_httpd.http_response(headers => array [omni_http.http_header('Content-Type', 'text/html')], body => 'test');
                                 http_response                                  
--------------------------------------------------------------------------------
 omni_httpd.http_response(("\\x74657374",200,"{""(Content-Type,text/html)""}"))
(1 row)

--- Shortcut syntax with body first
select omni_httpd.http_response('test');
                                             http_response                                              
--------------------------------------------------------------------------------------------------------
 omni_httpd.http_response(("\\x74657374",200,"{""(content-type,\\""text/plain; charset=utf-8\\"")""}"))
(1 row)

select omni_httpd.http_response('"test"'::json);
                                   http_response                                    
------------------------------------------------------------------------------------
 omni_httpd.http_response(("\\x227465737422",200,"{""(content-type,text/json)""}"))
(1 row)

select omni_httpd.http_response('"test"'::jsonb);
                                   http_response                                    
------------------------------------------------------------------------------------
 omni_httpd.http_response(("\\x227465737422",200,"{""(content-type,text/json)""}"))
(1 row)

